### PR TITLE
[PT-110] Retrieval BTC

### PR DESCRIPTION
**Pull request by 0xbuilder0**
# [PT-110](https://0xbase.atlassian.net/browse/PT-110) - Retrieval BTC

### Description

- Error Handling In BTC controller
- Update the BTC related structs
- Update the swagger docs
- Create an API the Get the BTC related data from tables 
- Update the postman collection
- Fix database relations and update the response for both btc and solana
- Create functions in wallet to get the BTC and solana data 
- Code refactor

[PT-110]: https://0xbase.atlassian.net/browse/PT-110?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: internal/routes/portfolio-router.go [hunk 1]
```diff
@@ -20,4 +20,6 @@ var PortfolioRoutes = func(router *gin.Engine, db *gorm.DB) {
 
 	v1.GET("/portfolio/btc/:btc-address", func(c *gin.Context) { controllers.BitcoinController(c, db) })
 
+	v1.GET("/portfolio/btc-wallet/:wallet-id", func(c *gin.Context) { controllers.GetBtcDataController(c, db) })
+
```
Comment (fuddin95) [inline internal/routes/portfolio-router.go:24]: why do we have two endpoints? 

`/portfolio/btc/:btc-address` ideally this endpoint should just return the wallet address information stored in the databse.

Comment (0xbuilder0) [reply internal/routes/portfolio-router.go:24]: You means that the one API gets the data from third party and save it into database and returns the data in response object

Comment (0xbuilder0) [reply internal/routes/portfolio-router.go:24]: changed and done

Comment (fuddin95) [reply internal/routes/portfolio-router.go:24]: yup

File: internal/controllers/bitcoin-controller.go [hunk 2]
```diff
@@ -21,61 +23,54 @@ import (
...
 // @Param        btc-address  path      string  true  "Bitcoin Address"
-// @Success      200          {object}  struct{ Data models.BitcoinAddressInfo "data"; ErrorCode int "error_code"; ErrNo int "err_no"; Message string "message"; Status string "status" }  "Returns wallet information including transaction history and balance"
-// @Failure      400          {object}  struct{ Error string }  "Bad Request"
-// @Failure      500          {object}  struct{ Error string }  "Internal Server Error"
-// @Router       /portfolio/btc/:btc-address [get]
-
-// BitcoinController handles requests for Bitcoin wallet information.
+// @Success      200 {object} models.GlobalWallet
+// @Failure      400 {object} errors.APIError
+// @Failure      404 {object} errors.APIError
```
Comment (fuddin95) [inline internal/controllers/bitcoin-controller.go:28]: duplicate, please fix in next PR

File: internal/controllers/bitcoin-controller.go [hunk 3]
```diff
@@ -99,32 +94,85 @@ func BitcoinController(c *gin.Context, db *gorm.DB) {
...
+// @Description  Retrieve BTC portfolio details, including BitcoinAddressInfo, for a specific wallet.
+// @Tags         bitcoin
+// @Accept       json
+// @Produce      json
+// @Param        wallet_id path int true "Wallet ID" Format(int)
+// @Param        offset query int false "Pagination offset" Format(int)
+// @Param        limit query int false "Pagination limit" Format(int)
+// @Success      200 {object} models.GlobalWallet
+// @Failure      400 {object} errors.APIError
+// @Failure      404 {object} errors.APIError
```
Comment (fuddin95) [inline internal/controllers/bitcoin-controller.go:136]: same here

**Review by fuddin95 — APPROVED** [review]

File: internal/controllers/bitcoin-controller.go (hunk 3 above)
Comment (0xbuilder0) [reply internal/controllers/bitcoin-controller.go:136]: the API response is same for if status code is 400, 404 or 500 etc that way added in-order to create the struct according to status code 
image

File: internal/controllers/bitcoin-controller.go (hunk 2 above)
Comment (0xbuilder0) [reply internal/controllers/bitcoin-controller.go:28]: same goes here

**Pull request merged**